src/pint/models/timing_model.py

```diff
@@ -1350,6 +1336,18 @@ def as_parfile(
 
         return result_begin + result_middle + result_end
 
+    def maskPar_has_toas_check(self, toas):
+        """Sanity check to ensure all maskParameters select at least one TOA."""
+        for maskpar in self.get_params_of_type_top("maskParameter"):
+            par = getattr(self, maskpar)
+            if "TNEQ" in str(par.name) or par.frozen:
+                continue
+            if par.is_mask and len(par.select_toa_mask(toas)) == 0:
+                raise AttributeError(
```

Comment:
@luojing1211 Since this function is only meant for `maskParameters`, and since I need to use `getattr()` to find out if it's frozen, shouldn't I immediately narrow it down to `maskParameters` with `self.get_params_of_type_top("maskParameter")`? Otherwise, I would have to iterate through every single parameter in the model to see if it's frozen.

Reply:
Get_params of type iterate through every single parameter as well. But it is a minor thing. We can fix it when we implementing a more robust sanity check.